File: springboot/open-api/src/main/java/com/openapi/domain/dto/ws/base/QueueMessage.java
package com.openapi.domain.dto.ws.base;

import lombok.Data;

/**
 * 队列消息包装
 */
@Data
public class QueueMessage<T> {
    private String id;
    private Long timestamp;
    private T payload;

    public QueueMessage(T payload) {
        this.id = java.util.UUID.randomUUID().toString();
        this.timestamp = System.currentTimeMillis();
        this.payload = payload;
    }
}

File: app/android/app/src/main/java/com/magicvector/domain/dto/ws/base/QueueMessage.kt
package com.magicvector.domain.dto.ws.base


import java.util.UUID

/**
 * 队列消息包装
 */
data class QueueMessage<T>(
    val id: String = UUID.randomUUID().toString(),
    val timestamp: Long = System.currentTimeMillis(),
    val payload: T
)